tapir/coop/tests/membership_resignation/test_form.py

```diff
@@ -21,30 +26,104 @@ def setUp(self) -> None:
         self.given_feature_flag_value(feature_flag_membership_resignation, True)
         mock_timezone_now(self, self.NOW)
 
-    def test_empty_form(self):
-        form = MembershipResignationForm()
-        self.assertIn("share_owner", form.fields)
-        self.assertIn("resignation_type", form.fields)
-        self.assertIn("cancellation_date", form.fields)
-        self.assertIn("transferring_shares_to", form.fields)
+    def test_isValid_sendingValidData_returnsTrue(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create()
+        data = {
+            "share_owner": share_owner,
+            "cancellation_reason": "A test reason.",
+            "cancellation_date": resignation.cancellation_date,
+            "resignation_type": resignation.resignation_type,
+            "transferring_shares_to": resignation.transferring_shares_to,
+            "paid_out": resignation.paid_out,
+        }
+        form = MembershipResignationForm(data=data)
+        self.assertTrue(form.is_valid())
 
-    def test_validations(self):
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_duplicates
+    def test_validate_share_owner_duplicateErrors_isThrown(self):
```

Comment:
The naming is a bit tricky for tests is a bit tricky, but I think this can be a good example:

test_functionBeingTested_caseBeingTested_expectedResult

Here, "functionBeingTested" is `validate_share_owner`, but we write it in camelCase so that it's easy to see the 3 parts in the function name. So "validateShareOwner"
Then, caseBeingTested is: the member already has a resignation. So we could do "memberIsAlreadyResigned" or "aResignationAlreadyExistsForThisMember".
Finally, "expectedResult": the function adds errors to the corresponding field. So "errorAddedToForm" or something similar?

That could give use `test_validateShareOwner_aResignationAlreadyExistsForThisMember_errorAddedToForm`. 
Hopefully that example can help you understand to desired naming. 
I won't comment on all test function names, let me know once you've reviewed them so I know when to check again.